src/main/java/de/uhd/ifi/se/decision/management/jira/rest/KnowledgeRest.java

```diff
@@ -1,108 +1,509 @@
+	 * 
+	 * @issue How can the sorting be implemented?
+	 *
+	 * @param id
+	 *            of the {@link KnowledgeElement} in database.
+	 * @param documentationLocation
+	 *            of the {@link KnowledgeElement}, e.g. "i" for Jira issue, see
+	 *            {@link DocumentationLocation}.
+	 * @param projectKey
+	 *            of the Jira project, see {@link DecisionKnowledgeProject}.
+	 * @return list of unlinked elements of the knowledge element for a project.
+	 *         Sorts the elements according to their similarity and their likelihood
+	 *         that they should be linked.
+	 */
+	@Path("/getUnlinkedElements")
+	@GET
+	@Produces({ MediaType.APPLICATION_JSON })
+	public Response getUnlinkedElements(@QueryParam("id") long id, @QueryParam("projectKey") String projectKey,
+			@QueryParam("documentationLocation") String documentationLocation) {
+		if (projectKey == null || id <= 0) {
+			return Response.status(Status.BAD_REQUEST).entity(ImmutableMap.of("error",
+					"Unlinked decision knowledge elements could not be received due to a bad request (element id or project key was missing)."))
+					.build();
+		}
+		KnowledgePersistenceManager persistenceManager = KnowledgePersistenceManager.getOrCreate(projectKey);
+		KnowledgeElement element = persistenceManager.getKnowledgeElement(id, documentationLocation);
+		List<KnowledgeElement> unlinkedDecisionKnowledgeElements = KnowledgeGraph.getOrCreate(projectKey)
+				.getUnlinkedElements(element);
+		return Response.ok(unlinkedDecisionKnowledgeElements).build();
+	}
+
+	/**
+	 * Creates a new {@link KnowledgeElement}. The decision knowledge element can
+	 * either be documented as a separate Jira issue (documentation location "i") or
+	 * in the description/a comment of an existing Jira issue (documentation
+	 * location "s").
+	 * 
+	 * @param request
+	 *            HttpServletRequest with an authorized Jira
+	 *            {@link ApplicationUser}.
+	 * @param element
+	 *            {@link KnowledgeElement} object with attributes, such as summary,
+	 *            description (optional), {@link DocumentationLocation}, and
+	 *            {@link KnowledgeType}.
+	 * @param idOfExistingElement
+	 *            optional parameter. Identifier of a parent element that the new
+	 *            element should be linked with. Either the id or the key needs to
+	 *            be passed, not both.
+	 * @param documentationLocationOfExistingElement
+	 *            optional parameter. Documentation location of a parent element
+	 *            that the new element should be linked with.
+	 * @param keyOfExistingElement
+	 *            optional parameter. Key of a parent element that the new element
+	 *            should be linked with. Either the id or the key needs to be
+	 *            passed, not both.
+	 * @return new {@link KnowledgeElement} with its internal database id set.
+	 */
+	@Path("/createDecisionKnowledgeElement")
+	@POST
+	@Produces({ MediaType.APPLICATION_JSON })
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'createDecisionKnowledgeElement(HttpServletRequest, KnowledgeElement, long, String, String)' has an NPath complexity of 432](https://app.codacy.com/manual/UHD/cures-condec-jira/pullRequest?prid=5419462)